### DISCUSSION TITLE
➤ YN0028: The lockfile would have been created by this install, which is explicitly forbidden

### DISCUSSION THREAD

**Discussion by dominicmarmionswrve** [callstack/react-native-builder-bob#534] [Q&A] [answered]
Hi,

I have created a new library via command `npx create-react-native-library myplugin`.
When running `yarn` or `yarn install` from the root of the library I get the following error:
```
➤ YN0028: The lockfile would have been created by this install, which is explicitly forbidden.
➤ YN0000: Failed with errors in 0s 3ms
```

My `yarn --version` is `3.6.1`

I've tried previous versions of `create-react-native-library` with same result so suspect this is an issue with my environment instead of a bug. Any help would be greatly appreciated!

Thanks.

**Comment by atlj** [comment, marked as answer]
Hey

I've found [this stack overflow issue](https://stackoverflow.com/questions/67062308/getting-yn0028-the-lockfile-would-have-been-modified-by-this-install-which-is-e) and they suggest running `yarn config get enableImmutableInstalls`, which outputs `false` for me and I don't have any issues installing packages.

I think you can try to set the ENV value as they are recommending on the SO issue.

**Reply by dominicmarmionswrve** [reply]
@atlj Thank you! - now why didn't I see that stack overflow post!! 🤦 

🍻

**Comment by PhpGuyZim** [comment]
"The lockfile would have been modified by this install, which is explicitly forbidden"
Spent a ridiculous amount of time trying to solve this on Heroku. Followed all the instructions to upgrade to Yarn 2, etc etc... then their system wouldn't accept by commits anymore. System was down the whole time.

**QUICK SOLUTION:** Delete/Move your **_yarn.lock_** and commit without it. Your commit will be accepted. Then you can restore your yarn.lock
Maybe one day I will have time to get to the actual source of the problem, but that day is not today.
